File: app/src/main/java/com/kkkkkn/readbooks/view/view/MainActivityView.java
package com.kkkkkn.readbooks.view.view;

import android.view.View;

import com.kkkkkn.readbooks.model.entity.BookInfo;

import java.util.ArrayList;

public interface MainActivityView extends BaseView {
    void syncBookShelf(ArrayList<BookInfo> list);
    void showUpdateDialog( String msg, String url, String version);
    void toSearchActivity();
    void toLoginActivity();
    void updateProgress(int progress);
    void installAPK(String filePath);
}

File: app/src/main/java/com/kkkkkn/readbooks/view/view/MainActivityView.kt
package com.kkkkkn.readbooks.view.view

import com.kkkkkn.readbooks.model.entity.BookInfo

interface MainActivityView :BaseView {
    fun syncBookShelf(list: ArrayList<BookInfo>)
    fun syncBookShelfError(str: String?)
    fun showUpdateDialog(msg: String, url: String, version: String)
    fun toSearchActivity()
    fun toLoginActivity()
    fun updateProgress(progress: Int)
    fun installAPK(filePath: String)
}